warehouse/query-core/src/main/java/datawave/query/jexl/JexlNodeFactory.java

```diff
@@ -108,142 +112,151 @@ public static JexlNode createNodeTreeFromFieldsToValues(ContainerType containerT
                     fieldsToValues.remove(identifier, value);
                 }
             }
-        } else {
-            JexlNodes.ensureCapacity(parentNode, fields.size());
         }
         
         for (String field : fields) {
-            ValueSet valuesForField = fieldsToValues.get(field);
+            JexlNode child = createNodeFromValuesForField(field, fieldsToValues, original, containerType, expandValues, isNegated);
+            if (child != null) {
+                JexlASTHelper.validateLineage(child, true);
+                children.add(child);
+            }
+        }
+        
+        switch (children.size()) {
+            case 0:
+                // in this case we had no matches for the range, so this expression gets replaced with a FALSE node.
+                return new ASTFalseNode(ParserTreeConstants.JJTFALSENODE);
+            case 1:
+                // Only one child node was generated, just return that
+                return children.get(0);
+            default:
+                // create the proper parent node and add children
+                JexlNode junction;
+                if (containerType.equals(ContainerType.OR_NODE)) {
+                    junction = new ASTOrNode(ParserTreeConstants.JJTORNODE);
+                } else {
+                    junction = new ASTAndNode(ParserTreeConstants.JJTANDNODE);
+                }
+                return JexlNodes.children(junction, children.toArray(new JexlNode[0]));
+        }
+    }
+    
+    /**
+     * Helper method for {@link JexlNodeFactory#createNodeTreeFromFieldsToValues(ContainerType, boolean, JexlNode, IndexLookupMap, boolean, boolean, boolean)}
+     *
+     * @param field
+     *            create a node for this field
+     * @param fieldsToValues
+     * @param original
+     *            the original JexlNode
+     * @param type
+     *            create an ASTOrNode or an ASTAndNode
+     * @param expandValues
+     *            should we expand values
+     * @param isNegated
+     * @return a JexlNode
+     */
+    public static JexlNode createNodeFromValuesForField(String field, IndexLookupMap fieldsToValues, JexlNode original, ContainerType type,
+                    boolean expandValues, boolean isNegated) {
+        
+        ValueSet valuesForField = fieldsToValues.get(field);
+        
+        if (!expandValues) {
             
-            // if not expanding values, then reuse the original node with simply a new field name (anyfield only)
-            if (!expandValues) {
-                JexlNode child = RebuildingVisitor.copy(orgNode);
-                for (ASTIdentifier identifier : JexlASTHelper.getIdentifiers(child)) {
-                    if (identifier.image.equals(Constants.ANY_FIELD)) {
-                        identifier.image = field;
-                    }
+            // If not expanding values set the original node's field name to '_ANYFIELD_'
```

Comment:
this looks like it replaced ANY_FIELD rather than setting it

Reply:
Thanks! Fixed in next commit.